arroyo-api/src/sources.rs

```diff
@@ -353,9 +368,17 @@ impl TryFrom<SchemaField> for SourceField {
             }
         };
 
+        let sql_name = match value.typ {
+            SchemaFieldType::Primitive(p) => primitive_to_sql(&p),
+            _ => ""
```

Comment:
I would list out the other cases here instead of using `_` so that it's more clear how we're handling each of them, and so that if add another SchemaFieldType we'll have to come here and hand that case.